airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlRdsSourcePerformanceSecretTest.java

```diff
@@ -0,0 +1,139 @@
+        .put("port", plainConfig.get("port"))
+        .put("database", dbName)
+        .put("username", plainConfig.get("username"))
+        .put("password", plainConfig.get("password"))
+        .put("replication_method", plainConfig.get("replication_method"))
+        .build());
+
+    final Database database = Databases.createDatabase(
+        config.get("username").asText(),
+        config.get("password").asText(),
+        String.format("jdbc:mysql://%s:%s/%s",
+            config.get("host").asText(),
+            config.get("port").asText(),
+            dbName),
+        "com.mysql.cj.jdbc.Driver",
+        SQLDialect.MYSQL,
+        "zeroDateTimeBehavior=convertToNull");
+
+    // It disable strict mode in the DB and allows to insert specific values.
+    // For example, it's possible to insert date with zero values "2021-00-00"
+    database.query(ctx -> ctx.execute("SET @@sql_mode=''"));
+    database.close();
+  }
+
+  @Test
+  public void test100tables100recordsDb() throws Exception {
+    int numberOfDummyRecords = 100; // 200 is near the max value for one shot in batching;
+    int numberOfColumns = 240;
+    int numberOfStreams = 100;
+    String schemaName = "test100tables100recordsDb";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void test1000tables240columns200recordsDb() throws Exception {
+    int numberOfDummyRecords = 200;
+    int numberOfColumns = 240;
+    int numberOfStreams = 1000;
+    String schemaName = "test1000tables240columns200recordsDb";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void test5000tables240columns200recordsDb() throws Exception {
+    int numberOfDummyRecords = 200;
+    int numberOfColumns = 240;
+    int numberOfStreams = 5000;
+    String schemaName = "test5000tables240columns200recordsDb";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void testSmall1000tableswith10000recordsDb() throws Exception {
+    int numberOfDummyRecords = 10001;
+    int numberOfColumns = 240;
+    int numberOfStreams = 1000;
+    String schemaName = "newsmall1000tableswith10000rows";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void testInterim15tableswith50000recordsDb() throws Exception {
+    int numberOfDummyRecords = 50010;
+    int numberOfColumns = 240;
+    int numberOfStreams = 15;
+    String schemaName = "newinterim15tableswith50000records";
+
+    setupDatabase(schemaName);
+
+    performTest(schemaName, numberOfStreams, numberOfColumns, numberOfDummyRecords);
+  }
+
+  @Test
+  public void testRegular25tables50000recordsDb() throws Exception {
+    int numberOfDummyRecords = 50003;
```

Comment:
Some of these numbers have slight differences. Is that on purpose? Not an issue. Just curious.

Reply:
Hi @tuliren. It depends on SQL script created by @andriikorotkov. As far as I understand - some PL\SQL differences between the DBs.